### ISSUE TITLE
Can't delete()  after save()

### ISSUE THREAD

**Issue by peekpt** [hivedb/hive#99]
**Code sample**
```dart
class Test extends HiveObject {
  int id;

  Test(this.id);
}

class TestAdapter extends TypeAdapter<Test> {
  @override
  Test read(BinaryReader reader) {
    return Test(reader.readInt());
  }

  @override
  void write(BinaryWriter writer, Test obj) {
    writer.writeInt(obj.id);
  }
}

main() async {
  Hive.init((await getApplicationDocumentsDirectory()).path);
  Hive.registerAdapter(TestAdapter(), 0);

  var box = await Hive.openBox<Test>('testBox');
  await box.clear();
  await box.add(Test(1));
  await box.add(Test(2));
  await box.add(Test(3));

  box.values.forEach((t) => print(t.id));

  box.values.first.id = 99;
  await box.values.first.save();

  box.values.forEach((t) => print(t.id));

  await box.values.first.delete(); // delete 99 entry

  box.values.forEach((t) => print(t.id));
}
```

console:

> I/flutter (32171): 1
I/flutter (32171): 2
I/flutter (32171): 3
I/flutter (32171): 99
I/flutter (32171): 2
I/flutter (32171): 3
I/flutter (32171): 99
I/flutter (32171): 2
I/flutter (32171): 3

**Comment by leisim** [comment]
Thanks for your great help! I found the problem and will publish a new version soon.

**Comment by leisim** [comment]
This will be fixed in `1.1.1` (published in a few minutes).
